elastic/apm-server
# magefile: fix and clean up packaging

## Motivation/summary

On beats main, `mage.UseElasticBeatPackaging` has been removed. For now we call the lower-level `mage.MustUsePackaging` directly.

Later when we look at [disabling cgo](https://github.com/elastic/apm-server/issues/6748), we should take more control over how and where we build our binaries. Then we can either use `mage.UseElasticBeatXPackPackaging`, which expects binaries in `build` rather than `x-pack/apm-server/build`, or else we should stop using the common Beats package templates.

Some cleanup while we're there:

 - remove Jaeger standalone config from apm-server.yml rendering
 - remove unused Mage targets
 - split Package target instead of using environment variables
 - stop building/distributing go-daemon, and System V init script

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~

## How to test these changes

Smoke test on RPM and Debian systems (install Deb/RPM, check that APM Server starts via systemd)

## Related issues

Closes https://github.com/elastic/apm-server/issues/7575
